SWLOR.Game.Server/Feature/ChatCommandDefinition/DMChatCommand.cs

```diff
@@ -890,5 +891,52 @@ private void Broadcast()
                     });
                 });
         }
+
+        private void SetScale()
+        {
+            const int MaxAmount = 50;
+
+            _builder.Create("setscale")
+                .Description("Sets an object's scale.")
+                .Permissions(AuthorizationLevel.DM, AuthorizationLevel.Admin)
+                .AvailableToAllOnTestEnvironment()
+                .RequiresTarget()
+                .Validate((user, args) =>
+                {
+                    // Missing an amount argument?
+                    if (args.Length <= 0)
+                    {
+                        return "Please specify the object's scale you want to set to. Valid range: 0.1-" + MaxAmount;
+                    }
+
+                    // Can't parse the amount?
+                    if (!float.TryParse(args[0], out var value))
+                    {
+                        return "Please specify a value between 0.1 and " + MaxAmount + ".";
+                    }
+
+                    // Amount is outside of our allowed range?
+                    if (value < 0.1 || value > MaxAmount)
+                    {
+                        return "Please specify a value between 0.1 and " + MaxAmount + ".";
+                    }
+
+                    return string.Empty;
+                })
+                .Action((user, target, location, args) =>
+                {
+                    // Allows the scale value to be a decimal number.
+                    float finalvalue = float.TryParse(args[0], out var value) ? value : 1;
```

Comment:
Additionally, you don't need to do a TryParse here because you've already validated it earlier. It should be safe to cast it to a float directly instead, but I don't feel strongly about changing it.